tests/test_deployments/test_app.py

```diff
@@ -310,6 +331,32 @@ def test_hard_reset(self, prune_fail: bool) -> None:
             assert response.status_code == 200
             assert data["status"] is not prune_fail
 
+    @wait_for_node_to_run
+    @pytest.mark.parametrize("dump_dir", [Path(tempfile.mkdtemp())])
+    def test_hard_reset_dev_mode(
+        self, dump_dir: Path, caplog: LogCaptureFixture
+    ) -> None:
+        """Test hard reset"""
+
+        resets = 0
+        period_dumper = PeriodDumper(logger=logging.getLogger(), dump_dir=dump_dir)
+        dump_dir = period_dumper.dump_dir
+
+        os.environ["ID"] = "_dummy_ID"
+        store_dir = dump_dir / f"period_{resets}"
+        path = self.tm_home / store_dir / ("node" + os.environ["ID"])
+        assert not path.exists()
+
+        with self.app.test_client() as client:
+            with mock.patch.object(app, "IS_DEV_MODE", return_value=True):
+                response = client.get("/hard_reset")
+                assert response.status_code == 200
+                data = cast(JSONLike, response.get_json())
+                assert data["status"] is True
+                assert "Dumped data for period" in caplog.text
+
+        assert path.exists()
+
```

Comment:
this is very interesting in fact. The OSError on the period dumper is only logged without the original error:
https://github.com/valory-xyz/open-autonomy/blob/6222f774e237c32c13c64e98c5271f6523463ac2/deployments/Dockerfiles/tendermint/app.py#L114-L127

We currently get a successful reset message returned, but the period dumping fails
```
INFO     deployments.Dockerfiles.tendermint.app:app.py:124 Error occurred while dumping data for period 0
```

Looking at the error, we find the following oddity: 
```
File name too long: '/tmp/tmp_17ez3hf/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0/node_dummy_ID/tm_state/period_0'
```
I cut it short there is in fact more of it. This seems to have to do with the fact that we catch the OSError in a nested try / except in which we also catch any Exception. Although not shown here the error is a `shutil.Error` which is raised in cpthon's `_copy_tree`, more precisely [here](https://github.com/python/cpython/blob/bf94c653f4291ba2db506453e0e00a82fe06b70a/Lib/shutil.py#L502-L514)